url82787aa7a4_vrthra_v_language_tgz-pJ8-v_VJ8/src/v/java/Helper.java

```diff
@@ -0,0 +1,212 @@
+            throw new VException("err:java:class_not_found", new Term<String>(v.Type.TString,c), "Java Exception : " + e.getMessage());
+        }
+    }
+
+    public static Method getMethod(String c, Term m, Quote params) {
+        String method = m.svalue();
+        try {
+            Class cls = getClass(c);
+            Class[] ptypes = getParamType(params);
+            return cls.getMethod(method, ptypes);
+        } catch (NoSuchMethodException e) {
+            throw new VException("err:java:no_such_method", new Term<String>(v.Type.TString,method), "Java Exception : " + e.getMessage());
+        }
+    }
+
+    public static Constructor getConstructor(String c, Quote params) {
+        try {
+            Class cls = getClass(c);
+            Class[] ptypes = getParamType(params);
+            return cls.getConstructor(ptypes);
+        } catch (NoSuchMethodException e) {
+            throw new VException("err:java:no_such_constructor",new Term<String>(v.Type.TString,c), "Java Exception : " + e.getMessage());
+        }
+    }
+
+    public static Field getField(String c, String method) {
+        try {
+            Class cls = getClass(c);
+            return cls.getField(method.substring(0, method.length() - 1));
+        } catch (NoSuchFieldException e) {
+            throw new VException("err:java:no_such_field",new Term<String>(v.Type.TString,method), "Java Exception : " + e.getMessage());
+        }
+    }
+
+    public static Term invoke(Term c, Term m, Quote params) {
+        String cname = null;
+        Object o = null;
+        String method = m.svalue();
+        if (c.type != v.Type.TSymbol) {
+            o = getJavaObj(c);
+            cname = o.getClass().getName();
+        } else {
+            cname = c.svalue();
+            o = null; // static invocation.
+        }
+
+        try {
+            if (method.equals("new")) {
+                if (o != null)
+                    throw new VException("err:java:invalid_class_name",c, "Java Exception invalid class");
+                Constructor cons = getConstructor(cname, params);
+                Object[] args = getParams(params);
+                return getResult(cons.newInstance(args));
+            } else if (method.endsWith("$")) {
+                // TODO: static field.
+                // Field.
+                Field fld = getField(cname, method);
+                // is it get or set?
+                int size = ((QuoteStream)params.tokens()).size();
+
+                if (size > 0) { // set
+                    Object no = getJavaObj(((QuoteStream)params.tokens()).get(0));
+                    fld.set(o, no);
+                    return getResult(no);
+                } else { // get
+                    return getResult(fld.get(o));
+                }
+            } else {
+                Method mtd = getMethod(cname, m, params);
+                Object[] args = getParams(params);
+                return getResult(mtd.invoke(o, args));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Catch `InvocationTargetException` explicitly when you call `Method.invoke()`. Use `Throwable.getCause()` or `getTargetException()` in the catch handler to access more information about the underlying exception. See the following diffs on GitHub that make this code change: [diff1](https://github.com/EsfingeFramework/gamification/commit/20792047e6c69457cca03e877f2ada578188f32a#diff-6808c12fe9ddaea042db84f95dd5ba78adab99d0cb18bb4d85916a8723530768L21), [diff2](https://github.com/cucumber-attic/cuke4duke/commit/f4821d400cf2590126f1298a8dff9f7a33d45dc5#diff-80adaffc20c0e6e7d2e42312386ffe1ddc10213bf5b8e8919895fd85d666c8deL34)

Learn more: [InvocationTargetException in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/InvocationTargetException.html)